ph5/utilities/set_deploy_pickup_times.py

```diff
@@ -85,91 +83,199 @@ def get_args():
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/ascii_s=%s\n" %
+                             pu_time[inc].ascii_s)
         else:
             of.write("\t%s\n" % line)
 
 
+def kefTables(kef_file, NICKNAME, PATH):
+
+    k = kefx.Kef(ARRAY_FILE)
+    k.open()
+    while True:
+        n = k.read(10000)
+        if n == 0:
+            LOGGER.error("Kef file is empty")
+            break
+        # Get Das_g references
+        ret = k.strip_receiver_g()
+        if ret:
+            add_references(ret)
+
+            # Make sure Array_t_xxx, Event_t_xxx, and Offset_t_aaa_sss exist
+            arrays, events, offsets = k.strip_a_e_o()
+
+
 def main():
-    global ARRAY_FILE, DEPLOY, PICKUP
+    global ARRAY_FILE, DEPLOY, PICKUP, AUTO_CORRECT, NICKNAME, PATH
 
     get_args()
-
+    dep_time = []
+    pu_time = []
     if not os.path.exists(ARRAY_FILE):
         LOGGER.error("Can't open {0}!".format(ARRAY_FILE))
         sys.exit()
     else:
         fh = open(ARRAY_FILE)
         mdir = os.path.dirname(ARRAY_FILE)
         base = os.path.basename(ARRAY_FILE)
-        base = '_' + base
+        base = 'autocortime_{0}'.format(base)
         of = open(os.path.join(mdir, base), 'w+')
-        LOGGER.info("Opened: {0}".join(os.path.join(mdir, base)))
-
-    dep_time = PH5_Time(passcal_s=DEPLOY)
-
-    pu_time = PH5_Time(passcal_s=PICKUP)
-
-    barf(fh, of, dep_time, pu_time)
-
+        # LOGGER.info("Opened: {0}".join(os.path.join(mdir, base)))
+    if AUTO_CORRECT:
+        ph5_api_object = ph5api.PH5(path=PATH, nickname=NICKNAME)
+        ph5_api_object.read_array_t_names()
+        if not ph5_api_object.Array_t_names:
+            LOGGER.error("No arrays or no events defined in ph5 file."
+                         "Can not continue!")
+            ph5_api_object.close()
+            sys.exit()
+        LOGGER.info("Writing updated kef file: {0}".format(of.name))
+        for array_name in ph5_api_object.Array_t_names:
+            ph5_api_object.read_array_t(array_name)
+            arraybyid = ph5_api_object.Array_t[array_name]['byid']
+            arrayorder = ph5_api_object.Array_t[array_name]['order']
+            for ph5_station in arrayorder:
+                station_list = arraybyid.get(ph5_station)
+                for deployment in station_list:
+                    station_len = len(station_list[deployment])
+                    for st_num in range(0, station_len):
+                        station = station_list[deployment][st_num]
+                        das = station['das/serial_number_s']
+                        true_deploy, true_pickup = \
+                            ph5_api_object.get_extent(
+                                das=station['das/serial_number_s'],
+                                component=station['channel_number_i'],
+                                sample_rate=station['sample_rate_i'])
+                        if type(true_deploy) != float:
```

Comment:
I wouldn't check for the type here, instead just check that `true_deploy` and `true_pickup` are not None. Because Python is a "duck typed" language, I believe that it is generally best practice to avoid explicitly checking the type.

i.e.
```python
if true_deploy is not None and true_pickup is not None:
   ... DO STUFF
```

Reply:
Commits 896cee4